### PR TITLE
removed init container for Postgres DB pod

### PR DESCRIPTION
**Pull request by dannyzaken**
### Explain the changes
* The init container was used to workaround an issue with hugepages which seems to be solved (see #661).
* After removing it, there is no need for a privileged container in Postgres DB scc

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1. See testing instructions [here](https://bugzilla.redhat.com/show_bug.cgi?id=1946792)

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: deploy/internal/statefulset-postgres-db.yaml [hunk 1]
```diff
@@ -19,92 +19,47 @@ spec:
...
-            memory: "500Mi"
-          limits:
-            cpu: "500m"
-            memory: "500Mi"
-        volumeMounts:
-        - name: db
-          mountPath: /var/lib/pgsql
-        - name: noobaa-postgres-initdb-sh-volume
-          mountPath: /init
       containers:
```
Comment (jackyalbo) [inline deploy/internal/statefulset-postgres-db.yaml:22]: I think removing all the init containers will result in this code failing as it expects init_containers[0] to exist:
https://github.com/noobaa/noobaa-operator/blob/6ddc25e52832366289685c4d813c35642b920c09/pkg/system/phase2_creating.go#L1537

**Comment by dannyzaken** [comment]
Converted back to draft. Still have some work to do with respect to DB upgrades

**Review by liranmauda — APPROVED** [review]
LGTM

File: pkg/system/phase2_creating.go [hunk 2]
```diff
@@ -244,20 +244,9 @@ func (r *Reconciler) SetDesiredNooBaaDB() error {
...
 	podSpec.SecurityContext.FSGroupChangePolicy = &defaultFSGroupChangePolicy
-	podSpec.InitContainers = util.FilterSlice(
-		podSpec.InitContainers,
-		func(c corev1.Container) bool {
-			// remove the init container that is not needed
-			return c.Name != "init"
-		},
-	)
+	// remove the init conatainer. It was used to workaround a hugepages issue, which was resolved in Postgres
+	podSpec.InitContainers = nil
```
Comment (jackyalbo) [inline pkg/system/phase2_creating.go:248]: Do we need this?

**Pull request merged**